### PR TITLE
Improve source generation performance

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type
What kind of change does this PR introduce?
- Refactoring

## What is the new behavior?
Xaml Source generation is performing faster, as more type resolutions are cached and less strings are used to interact with the Roslyn API. The update of the source generation tasks package also improves the incremental build performance using the new host server on windows.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

Internal Issue (If applicable):
https://nventive.visualstudio.com/DefaultCollection/Umbrella/_workitems/edit/143604

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.Reflection.cs [hunk 1]
```diff
@@ -0,0 +1,738 @@
...
+
+		private bool IsPrimitive(XamlObjectDefinition xamlObjectDefinition)
+		{
+			switch (xamlObjectDefinition.Type.Name)
+			{
+				case "Double":
+				case "Int32":
+				case "Int16":
+				case "Single":
+				case "Boolean":
```
Comment (dr1rrb) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.Reflection.cs]: What about  `Int64`, `UIntXX` and `Byte` (and other value types like `enum` and `char` ?)

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlRedirection/XamlMember.cs [hunk 2]
```diff
@@ -49,5 +50,20 @@ public bool IsAttachable
 			=> _declaringType != null ? _isAttachable : XamlConfig.IsUnoXaml ? _unoMember.IsAttachable : _msMember.IsAttachable;
 
 		public override string ToString() => XamlConfig.IsUnoXaml ? _unoMember.ToString() : _msMember.ToString();
+
+		public bool Equals(XamlMember other)
+			=> _name.HasValue()
+			? (_name == other._name && _declaringType == other._declaringType && _isAttachable == other.IsAttachable)
```
Comment (dr1rrb) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlRedirection/XamlMember.cs]: You should check if `other` is 'null'

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlRedirection/XamlType.cs [hunk 3]
```diff
@@ -38,5 +39,22 @@ public string PreferredXamlNamespace
 			=> _isUnknown ? unknownTypeNamespace : XamlConfig.IsUnoXaml ? _unoDeclaringType.PreferredXamlNamespace : _msDeclaringType.PreferredXamlNamespace;
 
 		public override string ToString() => XamlConfig.IsUnoXaml ? _unoDeclaringType.ToString() : _msDeclaringType.ToString();
+
+		public bool Equals(XamlType other) => _isUnknown
+			? other._isUnknown
```
Comment (dr1rrb) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlRedirection/XamlType.cs]: Same thing here `other` may be 'null'

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlRedirection/XamlMember.cs (hunk 2 above)
Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlRedirection/XamlMember.cs]: I surely home to remove this code once we can integrate the proper System.Xaml :)

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlRedirection/XamlType.cs (hunk 3 above)
Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlRedirection/XamlType.cs]: @carldebilly right!

**Pull request merged**